UnityProject/Assets/Scripts/HealthV2/Living/LivingHealthMasterBase.cs

```diff
@@ -828,8 +846,333 @@ public IEnumerator ProcessDNAPayload(DNAMutationData InDNAMutationData)
+						{
+							//yield return WaitFor.Seconds(1f);
+							var Mutation = BP.GetComponent<BodyPartMutations>();
+							if (Mutation != null)
+							{
+								if (string.IsNullOrEmpty(Payload.CustomisationTarget) == false || string.IsNullOrEmpty(Payload.CustomisationReplaceWith) == false)
+								{
+									Mutation.MutateCustomisation(Payload.CustomisationTarget,
+										Payload.CustomisationReplaceWith);
+								}
+
+								if (Payload.RemoveTargetMutationSO != null)
+								{
+									Mutation.RemoveMutation(Payload.RemoveTargetMutationSO);
+								}
+
+								if (Payload.TargetMutationSO != null)
+								{
+									Mutation.AddMutation(Payload.TargetMutationSO);
+								}
+
+								if (Payload.SpeciesMutateTo != null && Payload.MutateToBodyPart != null)
+								{
+									Mutation.ChangeToSpecies(Payload.SpeciesMutateTo,
+										Payload.MutateToBodyPart, characterSheet);
+									break;
+								}
+							}
+						}
+					}
+				}
+			}
+
+			//relooking for pumps because sometime old pumps wasn't removed when transferred to new part
+			reagentPoolSystem.PumpingDevices.Clear();
+
+			foreach (var x in BodyPartList)
+			{
+				if (x.TryGetComponent<Heart>(out var hrt))
+				{
+					reagentPoolSystem.PumpingDevices.Add(hrt);
+				}
+			}
+
+			//saving food
+			reagentPoolSystem.BloodPool.RemoveVolume(reagentPoolSystem.BloodPool.Total);
+			reagentPoolSystem.AddFreshBlood(reagentPoolSystem.BloodPool, reagentPoolSystem.StartingBlood);
+
+			var foodComps = GetSystem<HungerSystem>().NutrimentToConsume;
+
+			foreach (var x in foodComps)
+			{
+				if (blood.ContainsKey(x.Key))
+					reagentPoolSystem.BloodPool.Add(x.Key, blood[x.Key]);
+				else
+					reagentPoolSystem.BloodPool.Add(x.Key, 100);
+			}
+
+			//reagentPoolSystem.BloodPool.reagents = blood;
+			var bodyParts = new Dictionary<NamedSlot, ItemSlot>();
+			foreach (var x in playerScript.DynamicItemStorage.GetItemSlotTree())
+			{
+				if (x.NamedSlot != null && !bodyParts.ContainsKey((NamedSlot)x.NamedSlot))
+					bodyParts.Add((NamedSlot)x.NamedSlot, x);
+			}
 
+			foreach (var itemForPlace in itemsBeforeTransform)
+			{
+				if (itemForPlace.Item1 != null && bodyParts.ContainsKey(itemForPlace.Item2))
+				{
+					Inventory.ServerAdd(itemForPlace.Item1.gameObject, bodyParts[itemForPlace.Item2]);
+				}
+			}
+
+			if (dna != null && changeling != null)
+			{
+				var storage = changeling.ChangelingMind.CurrentPlayScript.DynamicItemStorage;
+
+				foreach (var id in dna.BodyClothesPrefabID)
+				{
+					var fakeClothes = Spawn.ServerPrefab(CustomNetworkManager.Instance.ForeverIDLookupSpawnablePrefabs[id]).GameObject;
+
+					ItemSlot bestSlot = storage.GetBestSlotFor(fakeClothes);
+
+					var placed = Inventory.ServerAdd(fakeClothes, bestSlot);
+
+					if (placed && (bestSlot.NamedSlot == NamedSlot.storage01 || bestSlot.NamedSlot == NamedSlot.storage02 || bestSlot.NamedSlot == NamedSlot.storage03
```

Comment:
I believe this can be simplified by just checking for every other named slot that isn't what you're targeting

Reply:
Maybe will be better something like this `((int?)bestSlot.NamedSlot) > 15.` Because I need all except storages and maybe handcuffs.

Reply:
yeah, that could work